include/dts-functions.sh

```diff
@@ -1387,3 +1387,70 @@ show_menu() {
     echo -ne "${RED}${SSH_OPT_UP}${NORMAL} to launch SSH server  ${NORMAL}"
   fi
 }
+
+check_if_fused() {
+file_path="/sys/class/mei/mei0/fw_status"
+
+if [[ ! -f $file_path ]]; then
+    echo "File not found: $file_path"
+    return 2
+fi
+
+hfsts6_value=""
+line_number=1
+while IFS= read -r line; do
+    if [[ $line_number -eq 6 ]]; then
+        hfsts6_value=$line
+        break
+    fi
+    ((line_number++))
+done < "$file_path"
+
+if [[ -z $hfsts6_value ]]; then
+    echo "Failed to read HFSTS6 value"
+    exit 1
+fi
+
+hfsts6_binary=$(echo "ibase=16; obase=2; $hfsts6_value" | bc)
+
+binary_length=${#hfsts6_binary}
+
+# Add leading zeros
+if [ $binary_length -lt 32 ]; then
+  padding=$((32 - binary_length))
+  zeros=$(printf "%${padding}s" | tr ' ' "0")
+  hfsts6_binary=$zeros$hfsts6_binary
+fi
+
+bit_30_value=${hfsts6_binary:1:1}
+
+if [ $bit_30_value == 0 ]; then
+  return 1
+else
+  return 0
+fi
+}
+
+check_if_boot_guard_enabled() {
+  # Get MSR in binary
+  msr=$(rdmsr 0x13a -0 | xxd -b | awk '{for(i=2; i<=NF; i++) if($i ~ /^[01]+$/) printf "%s", $i}' | tr -d '\n')
```

Comment:
Have you tested what happens if this MSR does not exist? E.g. on older Intel platforms such as Minnowboard, Dell Optiplex or PC Engines apu2?

Reply:
Checking the exit status should be enough -  If there is no MSR it returns a non-zero status code (It was `4` on APU, but I think it's safer to catch any non-zero status) https://github.com/Dasharo/dts-scripts/pull/15/commits/8bb0b85a8528df8e3bbbd7815ff9772ea6c6c18d#diff-1e8bf7e43e9fa22b80b3469478c83c53e8e7550daed419680e53bdcbff238b82R1436

Reply:
Not exactly. This BootGuard MSR exists on Intel platforms only, reading it on AMD should be skipped, and whole BootGuard logic as well.

Reply:
Do we have a mechanism in DTS that checks if the platform is Intel-based? The only thing I found is the `check_intel_regions` function, but it's probably not enough.

Reply:
You may read the `proc/cpuinfo` file. Intel processors have `vendor_id	: GenuineIntel` in the output

Reply:
d407c14bfb8efaccb8475b979738f1e46d6bffa6